### PR TITLE
Do get users with filters by params

### PR DESCRIPTION
**Pull request by burdinbogdan**
Rewrite some incorrect methods in users.
Change getting users by name using route
`/users/get/:username`
on getting users by filter params in route `/users` like
`/users/?role=admin`

### CODE REVIEW

File: src/v1/users/controllers/user-controller/index.js [hunk 1]
```diff
@@ -101,15 +87,14 @@ class UserController {
...
-        user.password = password;
-      }
-      if (role) {
-        user.role = role;
-      }
+
+      const filterData = { username, password, role };
+      const cleanFilterData = getCleanDataService(filterData);
+
+      Object.keys(cleanFilterData).forEach((key) => {
```
Comment (v1adko) [inline src/v1/users/controllers/user-controller/index.js]: Isn't this the same in getCleanData? Use that function

**Review by v1adko — CHANGES_REQUESTED** [review]

File: src/v1/users/controllers/user-controller/index.js (hunk 1 above)
Comment (burdinbogdan) [reply src/v1/users/controllers/user-controller/index.js]: Yes, it's similar,but not  same, getCleanData return data without undefined fields, and password to in basic view, but only in user model we transform password to salt and hash and this fields we need to save. So we need to do reassignment.
Or we can write `setCleanData(user, filterData)` for this case, but, how I understand, it's not clear for understanding way too.

Comment (burdinbogdan) [reply src/v1/users/controllers/user-controller/index.js]: Or write service what will do generate password hash and salt and use it in user model and here.

Comment (burdinbogdan) [reply src/v1/users/controllers/user-controller/index.js]: Hm, I will choose second variant. Are you agree?

File: src/lib/base-dao-mongoose.js [hunk 2]
```diff
@@ -41,7 +41,13 @@ class BaseDaoMongoose {
   }
 
   async updateById(_id, data) {
-    return Promise.resolve(this.Model.findOneAndUpdate({ _id }, data));
+    const item = await Promise.resolve(
+      this.Model.findOneAndUpdate({ _id }, data)
```
Comment (v1adko) [inline src/lib/base-dao-mongoose.js:45]: Should be lowercase everywhere. Uppercase variables are only constructor functions + classes

**Review by v1adko — APPROVED** [review]

**Comment by v1adko** [comment]
Rebase and fix it up

**Review by v1adko — CHANGES_REQUESTED** [review]
@burdinbogdan Also there is a problem with the tests on CI.
Fix it up. 
I want you to create a config file for jest (not in package json).
Put it in /internals/jest/* folder.
Also there is a setup script file option that specifies a file that runs before each test (until jest 22 comes out it's the only way for a global setup).

I want you to create __mocks__ folder there and mock logger.errror / logger info calls (assign it to noop or use jest.mock (better)).
It will avoid logger noise in testing. Then in tests you can assert on values using
`logger.info = jest.fn()`.

Sorry for missing this on the prev. PR, you can either do it all in a separate PR or fix it up here.

File: src/lib/base-dao-mongoose.js (hunk 2 above)
Comment (burdinbogdan) [reply src/lib/base-dao-mongoose.js:45]: Yes, it's mongoose class of model like User or Event.

Comment (v1adko) [reply src/lib/base-dao-mongoose.js:45]: But you're not using it as a constructor anywhere

**Comment by burdinbogdan** [comment]
@v1adko , Okay, I understand. Yes, if you don't mind I want do it in another PR. Can you give me reapprove for this PR?

File: src/lib/base-dao-mongoose.js (hunk 2 above)
Comment (burdinbogdan) [reply src/lib/base-dao-mongoose.js:45]: Yes, I use it as constructor only in inherits classes, like UserDao.
Something like:
```
class UserDao extends BaseDao {
  async create(username, password, role) {
    const user = new this.Model();
    user.username = username;
    ...
    super.create(user)
    ...
}
```

**Review by v1adko — APPROVED** [review]

**Comment by v1adko** [comment]
Open a new PR then :)

**Pull request merged**